specs/interop/predeploys.md

```diff
@@ -420,6 +430,151 @@ dependency set called `dependencySet()`. This function MUST return the array of
 `L1Block` MUST also provide a public getter to get the dependency set size called `dependencySetSize()`. This function
 MUST return the length of the dependency set array.
 
+## OptimismMintableERC20Factory
+
+| Constant | Value                                        |
+| -------- | -------------------------------------------- |
+| Address  | `0x4200000000000000000000000000000000000012` |
+
+### OptimismMintableERC20
+
+The `OptimismMintableERC20Factory` creates ERC20 contracts on L2 that can be used to deposit
+native L1 tokens into (`OptimismMintableERC20`). Anyone can deploy `OptimismMintableERC20` contracts.
+
+Each `OptimismMintableERC20` contract created by the `OptimismMintableERC20Factory`
+allows for the `L2StandardBridge` to mint
+and burn tokens, depending on whether the user is
+depositing from L1 to L2 or withdrawing from L2 to L1.
+
+### Updates
+
+The `OptimismMintableERC20Factory` is updated to include a `deployments` mapping
+that stores the `remoteToken` address for each deployed `OptimismMintableERC20`.
+This is essential for the liquidity migration process defined in the liquidity migration spec.
+
+### Functions
+
+#### `createOptimismMintableERC20WithDecimals`
+
+Creates an instance of the `OptimismMintableERC20` contract with a set of metadata defined by:
+
+- `_remoteToken`: address of the underlying token in its native chain.
+- `_name`: `OptimismMintableERC20` name
+- `_symbol`: `OptimismMintableERC20` symbol
+- `_decimals`: `OptimismMintableERC20` decimals
+
+```solidity
+createOptimismMintableERC20WithDecimals(address _remoteToken, string memory _name, string memory _symbol, uint8 _decimals) returns (address)
+```
+
+It MUST use `CREATE2` to deploy new contracts. The salt MUST depend on the following inputs:
+`_remoteToken`, `_name`, `_symbol`, and `_decimals`.
+This will ensure a unique `OptimismMintableERC20` for each set of ERC20 metadata.
+
+It MUST store the `_remoteToken` address for each deployed `OptimismMintableERC20` in a `deployments` mapping.
+
+A reference implementation looks like the following:
+
+```solidity
+function createOptimismMintableERC20WithDecimals(
```

Comment:
I think we could get away without implementing the whole function here and instead precisely define the salt in natural language and the invariant that the mapping must be updated

Reply:
perfect. Will also remove the reference implementation for the others